Uplift/Models/OpenHours.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  OpenHours.swift
+//  Uplift
+//
+//  Created by alden lamp on 9/19/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+
+struct OpenHours {
+
+    var rawHours: [Int: [QLOpenHours]]
+
+    var todaysHours: [QLOpenHours]? {
+        rawHours[Date().getIntegerDayOfWeekToday()]
+    }
+
+    var tomorrowsHours: [QLOpenHours]? {
+        rawHours[Date().getIntegerDayOfWeekTomorrow()]
+    }
+
+    init(openHours: [QLOpenHours]) {
+        rawHours = [:]
+        for openHour in openHours {
+            if rawHours.keys.contains(openHour.day) {
+                rawHours[openHour.day]?.append(openHour)
+            } else {
+                rawHours[openHour.day] = [openHour]
+            }
+        }
+    }
+
+    func isStatusChangingSoon(_ date: Date = Date()) -> Bool {
+        return todaysHours?.first { $0.willChangeSoon(date) == true } != nil
+    }
+
+    func isOpen(_ date: Date = Date()) -> Bool {
+        guard todaysHours != nil else {return false}
+        return !todaysHours!.filter { $0.isDateInRange(date) }.isEmpty
```

Comment:
do we need the two unwrappers ?

Reply:
what do you mean? Like the guard and then the force unwrap? I guess there's other ways to do it, what would you suggest?

Reply:
o wait I was confused with the !todaysHours! syntax my b